### ISSUE TITLE
too many open files

### ISSUE THREAD

**Issue by juev** [cloudflare/pint#69]
Hello, thx for linter!

We have repository with prometheus-alerts:

```bash
$ find conf -type f | wc -l
263
```

When we run linter we got error:
```bash
./pint --config rules.hcl lint conf/test
...
level=fatal msg="Execution completed with error(s)s" error="open conf/test/file.yml: too many open files"
```

**update**
if we run without config parameters, its work fine:
```bash
./pint lint conf/test
...
level=info msg="File parsed" path=conf/test/file.yml rules=1
```

**Comment by prymitive** [comment]
What's your `ulimit -n` output?
How many Prometheus servers do you have in your config?

**Comment by juev** [comment]
I use MacOS:
```bash
❯ ulimit -n
256
```

I dont use prometheus servers in config:
```hcl
rule {
  # Disallow spaces in label/annotation keys, they're only allowed in values.
  reject ".* +.*" {
    label_keys      = true
    annotation_keys = true
  }

  # Disallow URLs in labels, they should go to annotations.
  reject "https?://.+" {
    label_keys   = true
    label_values = true
  }

  # Check if all queries use correct vector matching rules
  vector_matching {}
}

rule {
  match {
    kind = "alerting"
  }

  # This block will apply to all alerting rules.
  # Each alert must have a 'summary' annotation on every alert.
  annotation "summary" {
    severity = "bug"
    required = true
  }

  # Each alert must have a 'grafana' annotation that links to grafana.
  annotation "grafana" {
    severity = "bug"
    value    = "(<|)http(s|)://(grafana.example.com)/(.+)"
  }

  # Each alert must have a 'wiki' annotation that links to wiki.
  annotation "wiki" {
    severity = "bug"
    value    = "(<|)https://(.+).example.com/(.+)"
  }

  # Each alert must have a 'severity' annotation that's either 'critical' or 'warning'.
  label "severity" {
    severity = "bug"
    value    = "(disaster|critical|high|warning|info)"
    required = true
  }

  # Check how many times each alert would fire in the last 1d.
  alerts {
    range   = "1d"
    step    = "1m"
    resolve = "5m"
  }

  # Check if templates used in annotations and labels are valid.
  template {}
}
```

**Comment by prymitive** [comment]
There was a missing file close, should be fixed via #71

**Comment by juev** [comment]
thx!
